Python updates 2022-03-02

mweinelt opened the pull request:
This is a major python update cycle that we do every once in a while. The last major update cycle started around mid-january (https://github.com/NixOS/nixpkgs/pull/154969).

https://hydra.nixos.org/job/nixpkgs/python-updates/tested#tabs-constituents

Things to watch out for:
- AWS packages (`awscli`, `boto3`, `botocore`, `moto`) package have very narrow version constraints
- Updates of `hypothesis` can be fragile for a huge number of package tests
  - 6.39.0 caused issues with pytest-asyncio, downgraded to 6.38.0
- `slackclient` before version 3.0, after that it's called `slack-sdk`, they're two distinct packages
  - false update dropped

Evaluations will appear shortly over at https://hydra.nixos.org/jobset/nixpkgs/python-updates.

The previous cycle can be found in #160067.

dotlambda committed:
python3Packages.pylama: 8.3.6 -> 8.3.7

mweinelt committed:
python3Packages.plotly: 5.5.0 -> 5.6.0

mweinelt committed:
python3Packages.platformdirs: 2.5.0 -> 2.5.1

mweinelt committed:
python3Packages.prompt-toolkit: 3.0.24 -> 3.0.28

mweinelt committed:
python3Packages.prometheus-client: 0.12.0 -> 0.13.1

mweinelt committed:
python3Packages.proto-plus: 1.19.8 -> 1.20.3

mweinelt committed:
python3Packages.prettytable: 3.0.0 -> 3.1.1

mweinelt committed:
python3Packages.poetry-core: 1.0.7 -> 1.0.8

mweinelt committed:
python3Packages.pooch: 1.5.2 -> 1.6.0

mweinelt committed:
python3Packages.pycognito: 2022.01.0 -> 2022.02.1

mweinelt committed:
python3Packages.proxy-py: 2.3.1 -> 2.4.0

mweinelt committed:
python3Packages.pyfakefs: 4.5.4 -> 4.5.5

mweinelt committed:
python3Packages.pyee: 8.2.2 -> 9.0.4

mweinelt committed:
python3Packages.pygit2: 1.8.0 -> 1.9.0

mweinelt committed:
python3Packages.pyfaidx: 0.6.3.1 -> 0.6.4

mweinelt committed:
python3Packages.pymemcache: 3.5.0 -> 3.5.1

mweinelt committed:
python3Packages.PyICU: 2.8 -> 2.8.1

mweinelt committed:
python3Packages.pylint-django: 2.5.0 -> 2.5.2

mweinelt committed:
python3Packages.pyomo: 5.7.3 -> 6.3.0

mweinelt committed:
python3Packages.pypdf3: 1.0.5 -> 1.0.6

mweinelt committed:
python3Packages.pyparsing: 3.0.6 -> 3.0.7

mweinelt committed:
python3Packages.pyperf: 2.3.0 -> 2.3.1

mweinelt committed:
python3Packages.pyopengl: 3.1.5 -> 3.1.6

mweinelt committed:
python3Packages.pyspark: 3.2.0 -> 3.2.1

mweinelt committed:
python3Packages.pyrsistent: 0.18.0 -> 0.18.1

mweinelt committed:
python3Packages.pyspnego: 0.3.1 -> 0.5.0

mweinelt committed:
python3Packages.pytesseract: 0.3.8 -> 0.3.9

mweinelt committed:
python3Packages.pytest-httpx: 0.17.3 -> 0.20.0

mweinelt committed:
python3Packages.pytest-mpl: 0.13 -> 0.14.0

mweinelt committed:
python3Packages.pytest-asyncio: 0.18.0 -> 0.18.1

mweinelt committed:
python3Packages.pytest-regressions: 2.3.0 -> 2.3.1

mweinelt committed:
python3Packages.pytest-mypy: 0.8.1 -> 0.9.1

mweinelt committed:
python3Packages.pytest-isort: 2.0.0 -> 3.0.0

mweinelt committed:
python3Packages.pytest-timeout: 2.0.2 -> 2.1.0

mweinelt committed:
python3Packages.pytest-runner: 5.3.1 -> 6.0.0

mweinelt committed:
python3Packages.python-cinderclient: 8.2.0 -> 8.3.0

mweinelt committed:
python3Packages.python3-saml: 1.12.0 -> 1.14.0

mweinelt committed:
python3Packages.pytest-socket: 0.5.0 -> 0.5.1

mweinelt committed:
python3Packages.pytest-testmon: 1.2.2 -> 1.3.0

mweinelt committed:
python3Packages.python-heatclient: 2.5.0 -> 2.5.1

mweinelt committed:
python3Packages.python-novaclient: 17.6.0 -> 17.7.0

mweinelt committed:
python3Packages.python-manilaclient: 3.2.0 -> 3.3.0

mweinelt committed:
python3Packages.python-ironicclient: 4.10.0 -> 4.11.0

mweinelt committed:
python3Packages.python-glanceclient: 3.5.0 -> 3.6.0

mweinelt committed:
python3Packages.python-dbusmock: 0.25.0 -> 0.26.1

mweinelt committed:
python3Packages.python-slugify: 6.1.0 -> 6.1.1

mweinelt committed:
python3Packages.pytools: 2021.2.9 -> 2022.1

mweinelt committed:
python3Packages.pytorch-lightning: 1.5.8 -> 1.5.10

mweinelt committed:
python3Packages.python-snappy: 0.6.0 -> 0.6.1

mweinelt committed:
python3Packages.pytorch-metric-learning: 1.1.0 -> 1.2.0

mweinelt committed:
python3Packages.python-swiftclient: 3.13.0 -> 3.13.1

mweinelt committed:
python3Packages.qiskit-machine-learning: 0.3.0 -> 0.3.1

mweinelt committed:
python3Packages.qiling: 1.4.1 -> 1.4.2

mweinelt committed:
python3Packages.pywbem: 1.4.0 -> 1.4.1

mweinelt committed:
python3Packages.pytorch-pfn-extras: 0.5.6 -> 0.5.7

mweinelt committed:
python3Packages.pywebview: 3.5 -> 3.6.1

mweinelt committed:
python3Packages.pyudev: 0.22.0 -> 0.23.2

mweinelt committed:
python3Packages.regex: 2022.1.18 -> 2022.3.2

mweinelt committed:
python3Packages.redis: 4.1.0 -> 4.1.4

mweinelt committed:
python3Packages.rasterio: 1.2.6 -> 1.2.10

mweinelt committed:
python3Packages.reportlab: 3.6.5 -> 3.6.8

mweinelt committed:
python3Packages.QtPy: 2.0.0 -> 2.0.1

mweinelt committed:
python3Packages.restructuredtext_lint: 1.3.2 -> 1.4.0

mweinelt committed:
python3Packages.requests-oauthlib: 1.3.0 -> 1.3.1

mweinelt committed:
python3Packages.sagemaker: 2.75.1 -> 2.77.1

mweinelt committed:
python3Packages.schema-salad: 8.2.20220103095339 -> 8.2.20220204150214

mweinelt committed:
python3Packages.s3fs: 2022.1.0 -> 2022.2.0

mweinelt committed:
python3Packages.simple-salesforce: 1.11.4 -> 1.11.6

mweinelt committed:
python3Packages.sip: 6.5.0 -> 6.5.1

mweinelt committed:
python3Packages.scikit-build: 0.12.0 -> 0.13.1

mweinelt committed:
python3Packages.scipy: 1.7.3 -> 1.8.0

mweinelt committed:
python3Packages.Shapely: 1.8.0 -> 1.8.1.post1

mweinelt committed:
python3Packages.smbprotocol: 1.8.3 -> 1.9.0

mweinelt committed:
python3Packages.spacy-transformers: 1.1.3 -> 1.1.4

mweinelt committed:
python3Packages.spyder: 5.2.1 -> 5.2.2

mweinelt committed:
python3Packages.superqt: 0.2.5-1 -> 0.3.1

mweinelt committed:
python3Packages.teletype: 1.1.0 -> 1.3.2

mweinelt committed:
python3Packages.sunpy: 3.1.3 -> 3.1.4

mweinelt committed:
python3Packages.stone: 3.2.1 -> 3.3.1

mweinelt committed:
python3Packages.symengine: 0.9.0 -> 0.9.1

mweinelt committed:
python3Packages.starlette: 0.17.1 -> 0.18.0

mweinelt committed:
python3Packages.tensorflow-estimator: 2.7.0 -> 2.8.0

mweinelt committed:
python3Packages.tensorflow-metadata: 1.5.0 -> 1.7.0

mweinelt committed:
python3Packages.testpath: 0.5.0 -> 0.6.0

mweinelt committed:
python3Packages.tempora: 5.0.0 -> 5.0.1

mweinelt committed:
python3Packages.terminado: 0.12.1 -> 0.13.1

mweinelt committed:
python3Packages.tensorflow-datasets: 4.4.0 -> 4.5.2

mweinelt committed:
python3Packages.trio: 0.19.0 -> 0.20.0

mweinelt committed:
python3Packages.tomlkit: 0.8.0 -> 0.10.0

mweinelt committed:
python3Packages.toggl-cli: 2.4.3 -> 3

mweinelt committed:
python3Packages.tqdm: 4.62.3 -> 4.63.0

mweinelt committed:
python3Packages.tpm2-pytss: 0.1.9 -> 1.0.0

mweinelt committed:
python3Packages.trytond: 6.2.3 -> 6.2.6

mweinelt committed:
python3Packages.towncrier: 21.3.0 -> 21.9.0

mweinelt committed:
python3Packages.tiledb: 0.12.0 -> 0.13.0

mweinelt committed:
python3Packages.unidiff: 0.7.0 -> 0.7.3

mweinelt committed:
python3Packages.ufo2ft: 2.25.2 -> 2.25.3

mweinelt committed:
python3Packages.twilio: 7.5.0 -> 7.7.0

mweinelt committed:
python3Packages.twine: 3.7.1 -> 3.8.0

mweinelt committed:
python3Packages.tweepy: 4.5.0 -> 4.6.0

mweinelt committed:
python3Packages.typing-extensions: 4.0.1 -> 4.1.1

mweinelt committed:
python3Packages.Twisted: 21.7.0 -> 22.1.0

mweinelt committed:
python3Packages.txaio: 21.2.1 -> 22.2.1

mweinelt committed:
python3Packages.websocket-client: 1.2.3 -> 1.3.1

mweinelt committed:
python3Packages.uvicorn: 0.16.0 -> 0.17.5

mweinelt committed:
python3Packages.unittest-xml-reporting: 3.0.4 -> 3.2.0

mweinelt committed:
python3Packages.virtualenv: 20.13.0 -> 20.13.2

mweinelt committed:
python3Packages.uproot: 4.1.9 -> 4.2.0

mweinelt committed:
python3Packages.xarray: 0.20.2 -> 2022.3.0

mweinelt committed:
python3Packages.yq: 2.13.0 -> 2.14.0

mweinelt committed:
python3Packages.werkzeug: 2.0.2 -> 2.0.3

mweinelt committed:
python3Packages.xxhash: 2.0.2 -> 3.0.0

mweinelt committed:
python3Packages.wsproto: 1.0.0 -> 1.1.0

mweinelt committed:
python3Packages.zarr: 2.10.3 -> 2.11.0

mweinelt committed:
python3Packages.zopfli: 0.1.9 -> 0.2.0

mweinelt committed:
python3Packages.zope.exceptions: 4.4 -> 4.5

mweinelt committed:
python3Packages.zimports: 0.4.1 -> 0.5.0

jonringer committed:
python3Packages.mongoengine: relax pymongo deps

jonringer committed:
python3Packages.lz4: remove stream import check, no longer exists

jonringer committed:
python3Packages.lz4: fix build and tests

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/1746506.html

- sphinx fixed by dropping docutils upgrade
- aiohttp fixed by disabling tests that trigger deprecation warnings in pytest 7

will trigger another eval shortly

mweinelt committed:
python3Packages.aiohttp: disable tests causing warnings with pytest>=7.0

mweinelt commented:
- Dropped the `astroid==2.10.0` bump to fix `pylint` and `asttokens`

mweinelt committed:
python3Packages.anyio: 3.3.4 -> 3.5.0

mweinelt commented:
- Fixed `httpcore` by relaxing h11 constraint and introducing `socksio`

mweinelt commented:
@DavHau Can you please check on scikit-learn?

jonringer committed:
python3Packages.httplib2: disable improperly marked tests

jonringer committed:
python3Packages.azure-mgmt-trafficmanager: fix build

jonringer committed:
python3Packges.azure-identity: relax msal-extensions bounds

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/1746580.html

jonringer commented:
@GrahamcOfBorg eval

mweinelt committed:
python3Packages.django-appconf: rename from django_appconf, clean up

mweinelt committed:
python3Packages.django-appconf: 1.0.3 -> 1.0.5

mweinelt committed:
python3Packages.promise: fix build

mweinelt committed:
python3Packages.ansi: fix build, enable tests

risicle commented:
Wait, scratch that - looks like it was just flakiness.

jonringer committed:
python3Packages.pybluez: mark disabled, requires use2to3

jonringer committed:
python3Packages.azure-mgmt-loganalytics: prune unneeded dependency

jonringer committed:
azure-cli: 2.32.0 -> 2.34.1

DavHau committed:
python3Packages.watchdog: fix build

mweinelt commented:
Will rebase as soon as hydra is back and trigger another eval. I think we're getting pretty close, but we still need to solve the `scikit-learn` problem.

SuperSandro2000 commented on pkgs/development/python-modules/ansi2html/default.nix:
```diff
@@ -2,13 +2,14 @@
+  version = "1.7.0";
+  format = "pyproject";
 
   disabled = !isPy3k;
```
At some point we should remove all of these.

SuperSandro2000 commented on pkgs/development/python-modules/trio/default.nix:
```diff
@@ -13,19 +13,45 @@
+    yapf
+  ];
+
+  preCheck = ''
```
```suggestion
  postPatch = ''
```

mweinelt replied:
Aha, what's the benefit?

SuperSandro2000 replied:
That all code patching is done in patches or postPatch and this isn't changing after building, installing or checking. Doing patching in multiple places is getting really messy the more things are changed especially when it is done in multiple phases.

mweinelt replied:
Agree to disagree.

SuperSandro2000 commented on pkgs/development/python-modules/rasterio/default.nix:
```diff
@@ -7,14 +7,14 @@
 buildPythonPackage rec {
   pname = "rasterio";
-  version = "1.2.6";
+  version = "4";
```
```suggestion
  version = "1.3a3";
```

mweinelt replied:
I have a fixup queued locally.

mweinelt replied:
I pushed an update to 1.2.10 that I tested on master.

SuperSandro2000 commented on pkgs/development/python-modules/pygame-gui/default.nix:
```diff
@@ -9,14 +9,14 @@
 buildPythonPackage rec {
   pname = "pygame-gui";
-  version = "0.6.4";
+  version = "064";
```
Please bring back the dots. Upstream advertises them with dots, only the tags are weird https://github.com/MyreMylar/pygame_gui/releases/tag/v_064

mweinelt replied:
Will drop that commit 689a2ecafd64c77dfd1b45d3f606d26420f5a71c in the next rebase.

DavHau committed:
python3Packages.scikit-learn: disable more tests

SuperSandro2000 commented on pkgs/development/python-modules/pybluez/default.nix:
```diff
@@ -2,11 +2,14 @@
   version = "unstable-20160819";
   pname = "pybluez";
+  # requires use2to3, which is no longer supported in setuptools>58
+  disabled = isPy3k;
```
```suggestion
  disabled = true;
```
we are always in python3 in this directory.

mweinelt replied:
How so? I can still (try and) build `python2Packages.pybluez`.

SuperSandro2000 replied:
Ah, you are right. I thought python2Packages wasn't overlayed https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/python2-packages.nix#L1-L3

DavHau commented:
> Will rebase as soon as hydra is back and trigger another eval. I think we're getting pretty close, but we still need to solve the `scikit-learn` problem.

I've been working on this today. See https://github.com/davhau/nixpkgs/tree/python-updates.
I disabled all tests related to `KernelPCA` and a few other ones.
I have no idea why these tests are failing, and some of them (the last few in the list) only fail sometimes.
The errors don't seem to be python exceptions and, I guess, are triggered by extensions.

jonringer committed:
python3Packages.meshio: add missing deps

jonringer commented:
I created a personal jobset here: https://hydra.jonringer.us/jobset/nixpkgs/python-updates#tabs-evaluations

jonringer committed:
python3Packages.botocore: 1.24.11 -> 1.24.12

jonringer committed:
python3Packages.boto3: 1.21.11 -> 1.21.12

jonringer committed:
awscli2: 2.4.19 -> 2.4.23

jonringer committed:
awscli: 1.22.35 -> 1.22.67

jonringer committed:
python310Packages.typeguard: disable python310 incompatible tests

risicle commented:
(just rebased on top of staging)

risicle commented:
Come to think, is rebasing or merging preferred in that kind of situation? I guess merging would allow signed commits to be retained.

jonringer committed:
python3Packages.plumbum: disable pytest==7.* incompatible tests

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/1746691.html

DavHau commented:
> > I disabled all tests related to KernelPCA and a few other ones.
> 
> @DavHau I don't think simply disabling the tests is really an option here. These are crashers, which suggest a serious problem with the built extension. It could potentially even be a security issue to allow it to be released like this.

Thanks for the information. I currently don't have the capacity to tinker any more with this.

mweinelt commented:
One more rebase and a few dropped commits.

mweinelt commented:
@risicle I think I prefer rebases because they create less noise and we can fixup broken commits instead of piling on new git objects.

mweinelt committed:
awscli2: prune dependency substitution for cryptography

mweinelt committed:
python3Packages.deepdiff: 5.6.0 -> 5.7.0

Relaxes the ordered-set constraint.

mweinelt committed:
python3Packages.tomli: 1.2.2 -> 2.0.1

https://github.com/hukkin/tomli/blob/2.0.1/CHANGELOG.md#201

mweinelt committed:
python3Packages.hatchling: init at 0.18.0

mweinelt committed:
python3Packages.devtools: ignore pytest8 deprecation warning

mweinelt committed:
python3Packages.pyownet: drop use_2to3 from setup.py

https://github.com/miccoli/pyownet/issues/12

mweinelt committed:
python3Packages.datadog: 0.43.0 -> 0.44.0

mweinelt committed:
python3Packages.chalice: 1.26.4 -> 1.26.6

mweinelt committed:
python3Packages.django_2: drop

The package goes EOL in 2022/04 and is therefore being removed.

Closes: #157905

mweinelt committed:
python3Packages.django-raster: drop django version condition

mweinelt committed:
home-assistant: update pytest-aiohttp and related overrides

mweinelt committed:
python3Packages.sanic-testing: relax httpx constraint

mweinelt committed:
python3Packages.sanic-testing: extract tests into passthru.pytest

Previously the pytestCheckHook had no meaning, since the tests were
always executed using by other means. Extracting the tests into a
dedicated package has become the standard way to escape infinite
recursion issues and offers a cleaner approach here.

mweinelt committed:
python3Packages.sanic: update dependency pins, disable failing test

mweinelt committed:
python3Packages.typing-inspect: switch to pytest and disable failing test

mweinelt committed:
python3Packages.sentry-sdk: disable failing tests

mweinelt committed:
python3Packages.twilio: migrate to pytest

mweinelt committed:
python3Packages.xmlsec: disable failing test

mweinelt committed:
python3Packages.plumbum: disable failing test

mweinelt committed:
python3Packages.metar: patch flaky test

mweinelt committed:
python3Packages.zeep: apply patch to fix pytest-httpx testcase

mweinelt committed:
python3Packages.imageio: fix tests

mweinelt committed:
home-assistant: pin hatasmota at 0.3.1

mweinelt committed:
python3Packages.pymongo: 3.12.2 -> 3.12.3

mweinelt committed:
python3Packages.cattrs: disable failing test

mweinelt committed:
python3Packages.cattrs: use pytest-xdist

Upstream uses it as well and it cuts the complete build down from
~14 minutes to ~1m30s.

mweinelt committed:
python3Packages.apsw: patch tests

mweinelt commented:
That last commit looks to be redundant with [`b5f2840` (#162538)](https://github.com/NixOS/nixpkgs/pull/162538/commits/b5f284063f84d8a58e3a8c0b60301684e23b23cd).

mweinelt committed:
platformio: relax dependency constraints

mweinelt committed:
python3Packages.tensorboard: rename from tensorflow-tensorboard, relax deps

mweinelt committed:
python3Packages.tensorboard: drop leftover 1.15 packaging

Couldn't find any remaining reference

mweinelt committed:
python3Packages.pytomlpp: disable tests incompatible with pytest7

mweinelt committed:
python3Packages.sqlalchemy: 1.4.31 -> 1.4.32

mweinelt committed:
python3Packages.aionotify: patch away loop kwarg in wait_for call

Fixes python3.10 compat

mweinelt committed:
python310Packages.tornado_5: disable

mweinelt committed:
python310Packages.pyuv: disable

SuperSandro2000 commented on pkgs/development/python-modules/tornado/5.nix:
```diff
@@ -3,12 +3,13 @@
   pname = "tornado";
   version = "5.1.1";
-  disabled = isPy27;
+  disabled = isPy27 || pythonAtLeast "3.10";
```
```suggestion
  disabled = pythonOlder "3.5" || pythonAtLeast "3.10";
```

mweinelt replied:
I would rather not readd support for python2 that way.

SuperSandro2000 replied:
I took that from https://pypi.org/project/tornado/ on the bottom left.

mweinelt replied:
Check again, this is tornado 5.x. https://pypi.org/project/tornado/5.1.1/

SuperSandro2000 replied:
Oh, you're right.

mweinelt committed:
python3Packages.hatchling: 0.18.0 -> 0.20.0

SuperSandro2000 commented on pkgs/development/embedded/platformio/core.nix:
```diff
@@ -153,7 +153,9 @@ with python.pkgs; buildPythonApplication rec {
-      --replace "zeroconf==0.37.*" "zeroconf"
+      --replace "wsproto==1.0.*" "wsproto" \
+      --replace "zeroconf==0.38.*" "zeroconf"
+
```
```suggestion
```
nit

SuperSandro2000 commented on pkgs/development/python-modules/tensorboard/default.nix:
```diff
@@ -23,15 +23,15 @@
 buildPythonPackage rec {
   pname = "tensorboard";
-  version = "2.6.0";
+  version = "2.8.0";
```
I think you added this by accident to the pyopengl commit.

SuperSandro2000 commented on pkgs/development/python-modules/sanic-testing/default.nix:
```diff
@@ -35,18 +34,19 @@ buildPythonPackage rec {
+  # check in passthru.tests.pytest to escape infinite recursion with sanic
+  doCheck = false;
+  doInstallCheck = false;
+  dontUsePythonImportsCheck = true;
```
I don't think we need this line if pythonImportsCheck is not set.

SuperSandro2000 commented on pkgs/development/python-modules/sentry-sdk/default.nix:
```diff
@@ -126,6 +126,9 @@ buildPythonPackage rec {
     "tests/integrations/rq/"
+    # broken since pytest 7.0.1; AssertionError: previous item was not torn down properly
+    "tests/utils/test_contextvars.py"
+
```
```suggestion
```
nit

SuperSandro2000 reviewed (commented):
some more or less usefull comments. The other things  are LGTM

mweinelt committed:
python310Packages.pyopengl-accelerate: disable

mweinelt committed:
python3Packages.tensorboard: 2.6.0 -> 2.8.0

mweinelt committed:
python3Packages.tensorflow: 2.7.1 -> 2.8.0

https://github.com/tensorflow/tensorflow/releases/tag/v2.8.0

mweinelt committed:
python3Packages.hatchling: 0.20.0 -> 0.20.1

Switches to the PyPi source per the upstream maintainers request in
https://github.com/NixOS/nixpkgs/pull/163088#issuecomment-1060748447

jonringer committed:
python310Packages.devtools: skip python 3.10 incompatible tests

jonringer committed:
python3Packages.numba: relax numpy version bounds

jonringer committed:
python3Packages.robotstatuschecker: fix src

mweinelt commented on pkgs/development/python-modules/robotstatuschecker/default.nix:
```diff
@@ -8,7 +8,7 @@ buildPythonPackage rec {
     owner = "robotframework";
     repo = "statuschecker";
-    rev = version;
+    rev = "refs/tags/v${version}";
```
Isn't that basically just `v${version}`? Or is that ambigious in that case?

jonringer replied:
not sure, I just used the bigger hammer

SuperSandro2000 replied:
Yes, it is and they have no conflicting branch with the same name.

jonringer committed:
python3Packages.pygit2: use libgit2 1.4.x

jonringer committed:
python3Packages.openapi-spec-validator: update build

jonringer committed:
python3Packages.jupyter_core: fix tests

jonringer committed:
python3Packages.lektor: fix src and build

jonringer committed:
python3Packages.python-daemon: disable obsolete version test

SuperSandro2000 commented on pkgs/development/python-modules/openapi-spec-validator/default.nix:
```diff
@@ -1,17 +1,23 @@
 buildPythonPackage rec {
   pname = "openapi-spec-validator";
   version = "0.4.0";
+  format = "pyproject";
```
```suggestion
  disabled = pythonOlder "3.7";
  format = "pyproject";
```

SuperSandro2000 commented on pkgs/development/python-modules/openapi-spec-validator/default.nix:
```diff
@@ -1,17 +1,23 @@
-  propagatedBuildInputs = [ jsonschema openapi-schema-validator pyyaml six setuptools ]
+  nativeBuildInputs = [ poetry-core ];
+
+  propagatedBuildInputs = [ jsonschema openapi-schema-validator pyyaml six setuptools requests ]
     ++ (lib.optionals (isPy27) [ pathlib ]);
```
```suggestion
  propagatedBuildInputs = [ jsonschema openapi-schema-validator pyyaml six setuptools requests ];
```

mweinelt committed:
python3Packages.pytorch: fix compatibility with recent setuptools versions

SuperSandro2000 commented on pkgs/development/python-modules/pytorch/default.nix:
```diff
@@ -135,7 +135,17 @@ in buildPythonPackage rec {
+        "test/run_test.py"
+      ];
+    })
+  ] ++ lib.optionals stdenv.isDarwin [
```
```suggestion
```

Otherwise we will probably never notice if the patch would be out of date.

mweinelt replied:
That patch was darwin-only before, I don't care about it.

mweinelt committed:
python3Packages.backports-zoneinfo: disable for python3.9 and newer

jonringer committed:
python3Packages.xgboost: limit tests to just basic tests

mweinelt committed:
python3Packages.jaraco_itertools: switch to pep518 build

jonringer commented:
@SuperSandro2000 build regressions are what we are really concerned about, if you have small fixes, feel free to just push them directory to this branch

mweinelt committed:
python3Packages.findpython: init at 0.1.3

mweinelt committed:
pdm: 1.12.6 -> 1.13.3

https://github.com/pdm-project/pdm/releases/tag/1.13.3
https://github.com/pdm-project/pdm/releases/tag/1.13.2
https://github.com/pdm-project/pdm/releases/tag/1.13.1
https://github.com/pdm-project/pdm/releases/tag/1.13.0.post0
https://github.com/pdm-project/pdm/releases/tag/1.13.0
https://github.com/pdm-project/pdm/releases/tag/1.12.8
https://github.com/pdm-project/pdm/releases/tag/1.12.7

mweinelt committed:
python3Packages.matrix-nio: relax h11 constraint

jonringer committed:
update-python-libraries.py: Always use ref/tags/ for specifying rev

Avoids situations where a branch and tag share the same
label. Github will silently return a non-valid artifact and
will fail the build.

Since all github releases correspond to an annotated tag, it's
safe to assume that if we found a release for a tag, that the
tag will exist.

jonringer committed:
python3Packages.adblock: fix src

mweinelt committed:
python3Packages.towncrier: let towncrier find its executable during tests

mweinelt committed:
python3Packages.flask-compress: fix build, enable tests

mweinelt committed:
python3Packages.commoncode: relax intbitset constraint

mweinelt committed:
python3Packages.dask: 2022.02.0 -> 2022.02.1

mweinelt committed:
python3Packages.nibabel: disable failing test

mweinelt committed:
python3Packages.doit: add toml dependency

mweinelt committed:
python3Packages.pooch: use pyproject format

mweinelt committed:
python3Packages.pyvcf: mark as broken

mweinelt committed:
python3Packages.ftfy: disable failing tests

mweinelt committed:
python3Packages.can: unstable-2022-01-11 -> 4.0.0

mweinelt committed:
python3Packages.pytools: fix build

jonringer committed:
python3Packages.adblock: update cargoSha

jonringer committed:
python3Packages.datatable: reduce test suite to small subset

test suite is very cpu intensive

mweinelt committed:
python3Packages.ciscoconfparse: propagate toml

SuperSandro2000 commented:
> @SuperSandro2000 build regressions are what we are really concerned about, if you have small fixes, feel free to just push them directory to this branch

Yeah, I would love to make sure that the openstack stuff still works but I currently really don't have the time to. 😞

jonringer committed:
python3Packages.qutip: don't run expensive test suite by default

jonringer committed:
python3Packages.pydmd: only run unit tests

Entire test suite can take 100+ vCPU hours

mweinelt committed:
python3Packages.cssselect2: build using flit-core

jonringer committed:
python3Packages.stumpy: limit default test suite to core tests

Full test suite is very CPU intensive

jonringer committed:
python3Packages.scikitimage: move CPU expensive test suite to passthru.tests

jonringer committed:
python310Packages.joblib: disable test which doesn't emit warning in python3.10

mweinelt committed:
python3Packages.pytest-pythonpath: drop

Obsoleted by https://github.com/pytest-dev/pytest/pull/9134, which was
released as part of pytest 7.0.0.

It has pytest<7 pinned, and we only have the single pytest version, so
it can be removed.

jonringer committed:
python3Packages.suds-jurko: mark broken

jonringer committed:
buildbot: disable sdist incompatible tests

jonringer committed:
python3Packages.openapi-core: disable sandbox incompatible tests

jonringer committed:
Revert "python3Packages.Shapely: 1.8.0 -> 1.8.1.post1"

This reverts commit ff0ad9ef441df73253ad29e767c4ed645cf2f477.

jonringer committed:
python3Packages.unittest-xml-reporting: add missing dependency

mweinelt committed:
python3Packages.stone: pull src from github

mweinelt committed:
python3Packages.clldutils: disable test that fail with pytest7

mweinelt committed:
python3Packages.clldutils: fix meta description and homepage

jonringer committed:
python3Packages.pyres: mark broken

mweinelt committed:
diffoscope: disable failing test

jonringer committed:
python3Packages.bsblan: fix build

jonringer committed:
python3Packages.superqt: fix build

SuperSandro2000 commented on pkgs/development/python-modules/stone/default.nix:
```diff
@@ -11,24 +12,23 @@ buildPythonPackage rec {
+    sed -i '/pytest-runner/d' setup.py
   '';
 
   nativeBuildInputs = [ pytest-runner ];
```
That could also be dropped if we remove the requirement.

mweinelt committed:
python3Packages.stone: drop pytest-runner

SuperSandro2000 committed:
python310Packages.osc-lib: disable broken tests

mweinelt committed:
python3Packages.pytorch: 1.10.2 -> 1.11.0

mweinelt committed:
python3Packages.pygls: propagate toml, use pyproject build

mweinelt committed:
python3Packages.teletype: fix build

mweinelt committed:
python3Packages.pytesseract: propagate packaging

mweinelt committed:
python3Packages.diff-cover: disable tests that fails with pytest7

mweinelt committed:
python3Packages.mutagen: refactor

- use pyproject build
- disable flaky test case
- prune dependencies
- create documentation output
- update meta

mweinelt committed:
python3Packages.tensorflow-metadata: fix build

SuperSandro2000 commented on pkgs/development/python-modules/pylama/default.nix:
```diff
@@ -45,14 +47,22 @@ buildPythonPackage rec {
 
   checkInputs = [
+    # avoid infinite recursion pylint -> isort -> pylama
+    (pylint.override {
+      isort = isort.overridePythonAttrs (old: {
+        doCheck = false;
+      });
+    })
```
Normally we put this kind of stuff into pythonPackages.

jonringer committed:
python3Packages.intensity-normalization.src: fix pname

jonringer committed:
python3Packages.bitcoinlib: fix openssl lib reference

mweinelt committed:
python3Packages.pytest-cid: relax pytest constraint

mweinelt committed:
python3Packages.tensorboardx: disable caffe2 tests

We don't have that dependency packaged.

mweinelt committed:
python3Packages.pynndescent: disable incompatible tests

mweinelt committed:
python3Packages.cython: 0.29.24 -> 0.29.28

mweinelt committed:
Revert "python3Packages.numpy: 1.21.5 -> 1.22.2"

This reverts commit 940af4b22772a13b2fe57e3f082e8d04634b5074.

Causes problems with scikit-learn, details can be found over in
https://github.com/NixOS/nixpkgs/pull/156578.

mweinelt committed:
Revert "python3Packages.scikit-learn: disable more tests"

This reverts commit 6cc7b4715feb572f53b37efef5abb6e35d256a76.

Disabling these tests hides crashing issues that are encountered in the
test suite. They are ultimately caused by an update to numpy>=1.22.0
that was reverted so that these tests can be reenabled.

mweinelt committed:
python3Packages.joblib: disable flaky test

mweinelt committed:
python3Packages.pgspecial: disable test that requires postgres instance

risicle commented:
Urgh `pytest-mpl` giving grief here, even once provided with jinja2 as it asks. Its tests get very meta and dependent on renderer details.

SuperSandro2000 commented:
@akaWolf ?

ofek commented:
How often does staging get merged into master?

jonringer commented:
> How often does staging get merged into master?

I try to help with staging, and I don't even know. There's a lot of potential blockers. On average though, every 2 weeks. give or take a week

ofek commented:
Got it, thanks!

friedelino commented:
hello! after upgrading `python3Packages.ipympl` to v0.8.8 with this merge build of this package fails because it lacks `matplotlib` as buildinput. i already opened a pr (#166979) to this.

Shados commented:
@mweinelt The updated levenshtein version fails to build, and that shows on the hydra jobs for it... why was it included in the merge?

mweinelt commented:
> why was it included in the merge?

Because this is a bulk update run, and we're trying our best to keep the package set stable and up to date. That doesn't work out in all cases sadly.

Let's continue working on this in #168547 or revert it that fails.

NorfairKing commented:
Looks like this PR was relevant to an issue that I've had: https://github.com/nix-community/poetry2nix/issues/605 
I've made an issue about backporting it: https://github.com/NixOS/nixpkgs/issues/171447

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/python3-11-tensorflow-2-12-how-to/30156/2
